src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (x) => {
+    if (x === undefined) {
```

Comment:
The check for 'undefined' should be strict. Use 'typeof x === "undefined"' to ensure that the check is not coerced, which can lead to unexpected behavior.